src/librustc_back/target/mod.rs

```diff
@@ -62,6 +62,85 @@ mod netbsd_base;
 mod windows_base;
 mod windows_msvc_base;
 
+macro_rules! supported_targets {
+    ( $(($triple:expr, $module:ident)),+ ) => (
+        /// List of supported targets
+        pub const TARGETS: &'static [&'static str] = &[$($triple),*];
+
+        // this would use a match if stringify! were allowed in pattern position
+        fn load_specific(target: &str) -> Option<Target> {
+            $(mod $module;)*
+            let target = target.replace("-", "_");
+            if false { }
+            $(
+                else if target == stringify!($module) {
+                    let t = $module::target();
+                    debug!("Got builtin target: {:?}", t);
+                    return Some(t);
+                }
+            )*
+            else if target == "x86_64-w64-mingw32" {
```

Comment:
While copy pasting this code, I noticed these last two else-if branches are effectively unreachable (because of the `let target = target.replace("-", "_");` a few lines above).

Not sure if I should fix them or just remove them. Probably the latter because @brson said these targets are deprecated.